Refresh schemas on datastore when retrieving them

arjansh opened the pull request:
Fixes #1461.

Make sure schemas on a datastore are refreshed when retrieving them from the datastores map on the ExcelOutputWriterFactory class, because they may contain stale references. This can particularly be the case when accessing multiple sheet within an Excel sheet within one job where existing sheets are overwritten, which in practice means a sheet is deleted and should be recreated, which obviously may fail if it is still being referenced even though it has been deleted.

LosD commented:
I'd like to see a little bit of JavaDoc on the `MultiStreamTransformer`. Just to tell its purpose (which is only testing, I guess)

LosD commented:
Otherwise LGTM

arjansh commented:
@LosD:
I added a javadoc to the `MultiStreamTransfomer`. Maybe it would also make sense to rename it to something like `MultiStreamTestTransformer`.

Next to that, the `testConcurrentWriting` test which I added to the CreateExcelSpreadsheetAnalyzerTest class is really an integration test and not a unit test. The fix for the test is even in another maven module. Maybe it should be moved and run as an integration test?

LosD commented:
Makes sense!

LosD commented on components/writers/src/test/java/org/datacleaner/extension/output/MultiStreamTestTransformer.java:
```diff
@@ -0,0 +1,82 @@
+        }
+        _even = !_even;
+    }
+}
```
Missing EOL

LosD commented:
Single EOL line missing, otherwise LGTM.

Might be easiest just to let Travis complete now, add the EOL and merge straight away. I'm more than confident that the newline wont break the build :)
